playbooks/prepare_host.yaml

```diff
@@ -148,13 +144,29 @@
           After=syslog.target
           [Service]
           Type=oneshot
-          ExecStart=/bin/bash -c '/sbin/losetup /dev/loop1 || \
-          /sbin/losetup /dev/loop1 /var/lib/ceph-osd.img ; partprobe /dev/loop1'
+          ExecStart=/bin/bash -c '/sbin/losetup /dev/loop1 || /sbin/losetup --direct-io=on /dev/loop1 /var/lib/ceph-osd.img'
           ExecStop=/sbin/losetup -d /dev/loop1
           RemainAfterExit=yes
           [Install]
           WantedBy=multi-user.target
+      register: create_losetup_service
     - name: Enable ceph-osd-losetup.service
       systemd:
         name: ceph-osd-losetup.service
         enabled: true
+        state: "{{ create_losetup_service.changed | ternary('restarted', 'started') }}"
+        daemon_reload: "{{ create_losetup_service.changed }}"
+    - name: Create volume group for ceph
+      lvg:
+        vg: vg_ceph
+        pvs: /dev/loop1
+    - name: Create ceph data LV
+      lvol:
+        vg: vg_ceph
+        lv: data
+        size: 96%VG
+    - name: Create ceph db LV
+      lvol:
+        vg: vg_ceph
+        lv: db
+        size: 4%VG
```

Comment:
this won't work, I tested it when implementing this thing and you need to keep space for some metadatas I think. Which is why I initially did `(ceph_loop_device_size * 4 / 100) - 0.1`

Reply:
It does work. I'm using it.